tensorboard/plugins/debugger/events_writer_manager_test.py

```diff
@@ -0,0 +1,435 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests the Tensorboard debugger data plugin."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import json
+import os
+import shutil
+import tempfile
+
+import mock
+import numpy as np
+import tensorflow as tf
+from werkzeug import test as werkzeug_test
+from werkzeug import wrappers
+
+# pylint: disable=ungrouped-imports, wrong-import-order
+from google.protobuf  import json_format
+from tensorboard.backend import application
+from tensorboard.backend.event_processing import event_multiplexer
+from tensorboard.plugins import base_plugin
+from tensorboard.plugins.debugger import constants
+from tensorboard.plugins.debugger import debugger_plugin
+from tensorboard.plugins.debugger import debugger_server_lib
+from tensorboard.plugins.debugger import numerics_alert
+from tensorflow.core.debug import debugger_event_metadata_pb2
+# pylint: enable=ungrouped-imports, wrong-import-order
+
+
+class DebuggerPluginTest(tf.test.TestCase):
+
+  def setUp(self):
```

Comment:
This large `setUp` method, and much of the rest of this class, is duplicated across files. So: (1) all of my comments on the other file apply here, and (2) what do you think about extracting it to a base `DebuggerTestCase(tf.test.TestCase)` class? At a glance, it looks like hundreds of lines, so it sounds worth it to me.